stix/idl/structures/stx_make_map.pro

```diff
@@ -0,0 +1,123 @@
+;+
+;
+; NAME:
+; 
+;   stx_make_map
+;
+; PURPOSE:
+; 
+;   Create a map from a STIX reconstruction (SOLO HPC coordinate). Observed visibilities used to reconstruct the map
+;   and visibilities predicted from the map are saved in the map structure
+;
+; CALLING SEQUENCE:
+; 
+;   stx_map = stx_make_map(im_map, aux_data, pixel, method, vis)
+;
+; INPUTS:
+; 
+;   im_map: image reconstruted with a STIX imaging method (STIX coordinate frame)
+;   
+;   aux_data: auxiliary data structure containing information on STIX pointing and spacecraft ephemeris
+;   
+;   pixel: bi-dimensional array containing the pixel size in arcsec
+;   
+;   method: string indicating which method has been used for reconstructing the image passed in im_map
+;   
+;   vis: visibility structure used for reconstructing an image
+;
+; OUTPUTS:
+; 
+;   Map structure. The following fields are added:
+;   
+;     - ENERGY_RANGE: array containing the lower and upper edge of the energy interval used for reconstructing 'im_map' (keV)
+;     
+;     - OBS_VIS: visibility structure used for reconstructing 'im_map'
+;     
+;     - PRED_VIS: structure containing the complex values of the visibilities predicted from 'im_map'. Used for displaying data 
+;                 fitting plots and for computing the chi2 value associated with 'im_map'
+;                 
+;     - AUX_DATA: auxiliary data srtucture used for reconstructing 'im_map'
+;     
+;     - RSUN: apparent radius of the SUN (arcsec)
+;     
+;     - L0: Heliographic longitude (degrees)
+;     
+;     - B0: Heliographic latitude (degrees)
+;              
+;     - COORD_FRAME: string indicating the coordinate system of the output map (SOLO HPC)
+;
+;     - UNITS: string indicating the units of reconstructed map (photons cm^-2 asec^-2 s^-1 keV^-1)
+;     
+; HISTORY: September 2022, Massa P., created
+;
+; CONTACT:
+;   paolo.massa@wku.edu
+;-
+
+function stx_make_map, im_map, aux_data, pixel, method, vis
```

Comment:
What the frame of the input stx_rtn? maybe an option to  not rotate

Reply:
The input image is conceived in the STIX coordinate frame (see line 18). I would not add the option not to rotate for now, it needs to be discussed more